### ISSUE TITLE
Missing similar methods for annotated sparse matrices

### ISSUE THREAD

**Issue by Sacha0** [JuliaLang/julia#19016]
`similar` appears to lack three-argument methods for annotated sparse matrices:

``` julia
julia> VERSION
v"0.6.0-dev.987"

julia> typeof(similar(Symmetric(speye(4)), Float64, (4,4)))
Array{Float64,2}

julia> @which(similar(Symmetric(speye(4)), Float64, (4,4)))
similar{T,N}(a::AbstractArray, ::Type{T}, dims::Tuple{Vararg{Int64,N}}) at abstractarray.jl:422

julia> # abstractarray.jl:422 returns an Array

julia> typeof(similar(Hermitian(speye(4)), Float64, (4,4)))
Array{Float64,2}

julia> @which(similar(Hermitian(speye(4)), Float64, (4,4)))
similar{T,N}(a::AbstractArray, ::Type{T}, dims::Tuple{Vararg{Int64,N}}) at abstractarray.jl:422

julia> typeof(similar(LowerTriangular(speye(4)), Float64, (4,4)))
Array{Float64,2}

julia> @which(similar(LowerTriangular(speye(4)), Float64, (4,4)))
similar{T,N}(a::AbstractArray, ::Type{T}, dims::Tuple{Vararg{Int64,N}}) at abstractarray.jl:422
```

Best!

**Comment by andreasnoack** [comment]
Should probably call `similar` on the underlying array.

**Comment by fredrikekre** [comment]
Fixed by #24163 and #24162

**Comment by Sacha0** [comment]
Hah, I'd forgotten filing this issue! Thanks for closing this `@fredrikekre`! :)
